### DISCUSSION TITLE
Can tqdm show progress for a file save operation?

### DISCUSSION THREAD

**Discussion by ooofest** [tqdm/tqdm#1418] [Q&A]
Hi, novice Python user here.

I've created a script which takes in a list of image files and generates a series of grid images based on a logical matrix of those original inputs, with optional x/y axis labels, etc.  There is a max size each grid image can reach, at which point the current one will be saved and a new one will be created with input files not yet processed.  It's using the Python PIL library for image creation, combining and saving.

I'm using tqdm to show progress on processing logic that combines the input image files for each grid (because those are for loops and have defined ranges), but the relatively long and undefined wait time for each file to be saved has no such display of progress - it just hangs until done:

https://pillow.readthedocs.io/en/stable/_modules/PIL/Image.html#Image.save

Again, being a novice with Python, it seemed the only thing I read which might possibly be related to showing progress for the PIL image save activity might be where tqdm can be piped for I/O operations.  But maybe that doesn't apply in this case for my script?

My code for saving is pretty simple:

image.save(image_filename)

Still, I thought it might be helpful to ask this question, as I didn't find similar questions yet in my searching.

**Comment by peterjc** [comment]
In theory you could show the number of bytes in ``image_filename`` (which could be a progress bar if you know the expected filesize in advance, which you probably can do for a fixed size BMP or similar, but probably can't for anything with compression like PNG or JPEG).

However, I would expect that the bulk of the time taken by ``image.save(image_filename)`` is computing the image, with writing to disk being the final and relatively quick part.

i.e. I don't think this is possible without help/changes to the PIL function, for example adding some call back functionality to report on progress.
